File: mobile/ios/benight-obj-c/benight-obj-c/ConnectViewController.m
//
//  ViewController.m
//  benight-obj-c
//
//  Created by Kevin VACQUIER on 15/02/2016.
//  Copyright © 2016 Kevin VACQUIER. All rights reserved.
//

#import "ConnectViewController.h"

@interface ConnectViewController ()

@end

@implementation ConnectViewController

- (void)viewDidLoad {
    [super viewDidLoad];
    // Do any additional setup after loading the view, typically from a nib.
}

- (void)didReceiveMemoryWarning {
    [super didReceiveMemoryWarning];
    // Dispose of any resources that can be recreated.
}

-(void)ErrorPopup:(NSString*)message {
    UIAlertController *alert = [UIAlertController alertControllerWithTitle:@"Error"
                                                                   message:message
                                                            preferredStyle:UIAlertControllerStyleAlert];
    
    UIAlertAction* ok = [UIAlertAction
                         actionWithTitle:@"OK"
                         style:UIAlertActionStyleDefault
                         handler:^(UIAlertAction * action)
                         {
                             [alert dismissViewControllerAnimated:YES completion:nil];
                             
                         }];
    [alert addAction:ok];
    
    [self presentViewController:alert animated:YES completion:nil];
    
    
}

- (IBAction)Connect {
    if ([self.idField.text length] > 0) {
        if ([self.passwdField.text length] > 0) {
            [PFUser logInWithUsernameInBackground:self.idField.text password:self.passwdField.text
                                            block:^(PFUser *user, NSError *error) {
                                                if (user) {
                                                    [self ErrorPopup:@"Login OK"];
                                                } else {
                                                    [self ErrorPopup:@"Login Failed"];
                                                }
                                            }];
        }
    }
}

@end

File: iOS/Benight/Benight/ConnectViewController.swift
	//
//  ViewController.swift
//  Benight
//
//  Created by Kévin VACQUIER on 18/11/2014.
//  Copyright (c) 2014 Benight. All rights reserved.
//

import UIKit

class ConnectViewController: ResponsiveTextFieldViewController {
	@IBOutlet var emailField: UITextField!
	@IBOutlet var passwdField: UITextField!
	
    func ErrorPopup(message: String)
    {
        let alertController = UIAlertController(title: "Error", message: message, preferredStyle: .Alert)
        
        let defaultAction = UIAlertAction(title: "OK", style: .Default, handler: nil)
        alertController.addAction(defaultAction)
        
        presentViewController(alertController, animated: true, completion: nil)
    }
    
    override func viewDidLoad()
	{
		super.viewDidLoad()
		UIApplication.sharedApplication().setStatusBarStyle(UIStatusBarStyle.LightContent, animated: true)
		
		self.view.backgroundColor = UIColor(patternImage: UIImage(named: "background")!)
		// Do any additional setup after loading the view.
		var tap:UITapGestureRecognizer = UITapGestureRecognizer(target: self,	action: "DismissKeyboard")
		view.addGestureRecognizer(tap)
	}
	
	override func didReceiveMemoryWarning() {
		super.didReceiveMemoryWarning()
		// Dispose of any resources that can be recreated.
	}
	
	
	func DismissKeyboard(){
		//Causes the view (or one of its embedded text fields) to resign the first responder status.
		view.endEditing(true)
	}
	
	@IBAction func ConnectUser(sender: UIButton)
	{
		
		if (count(emailField.text) > 0)
		{
			if (count(passwdField.text) > 0)
			{
                SwiftSpinner.show("Connection...")
				PFUser.logInWithUsernameInBackground(emailField.text, password: passwdField.text) {
					(user: PFUser?, error: NSError?) -> Void in
					if user != nil {
						print("login ok")
                        SwiftSpinner.hide()
						self.performSegueWithIdentifier("Connected", sender: nil)
					} else {
						self.ErrorPopup("Login Failed")}
                    SwiftSpinner.hide()
				}
			}
			else {
				ErrorPopup("Please Enter an Email")}
			
		}
		else {
			ErrorPopup("Please Enter an Email")}
	}
	
	@IBAction func ConnectWithFace(sender: AnyObject)
	{
        SwiftSpinner.show("Connection...")
		let permissions = ["user_about_me", "user_relationships", "user_birthday", "user_location"]
		PFFacebookUtils.logInWithPermissions(permissions,
			block: {
				(user: PFUser?, error: NSError?) -> Void in
				if user == nil
				{
                    SwiftSpinner.hide()
					NSLog("Uh oh. The user cancelled the Facebook login.")
				}
				else if user!.isNew
				{
                    SwiftSpinner.hide()
                    self.performSegueWithIdentifier("Connected", sender: nil)
				}
				else
				{
                    SwiftSpinner.hide()
                    self.performSegueWithIdentifier("Connected", sender: nil)
				}
		})
	}
}


